src/main/java/com/goncharenko/currencyexchangerate/service/impl/BankServiceImpl.java

```diff
@@ -0,0 +1,111 @@
+import com.goncharenko.currencyexchangerate.dto.BankDTO;
+import com.goncharenko.currencyexchangerate.dto.CurrencyDTO;
+import com.goncharenko.currencyexchangerate.exceptions.ResourceNotFoundException;
+import com.goncharenko.currencyexchangerate.service.BankService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.util.CollectionUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static com.goncharenko.currencyexchangerate.dto.CurrencyDTO.convertToDTO;
+import static com.goncharenko.currencyexchangerate.dto.CurrencyDTO.convertToDomain;
+
+@Component
+public class BankServiceImpl implements BankService {
+    private static final Logger LOGGER = LoggerFactory.getLogger(BankServiceImpl.class);
+    private final BankRepository bankRepository;
+    private final CurrencyRepository currencyRepository;
+
+    public BankServiceImpl(BankRepository bankRepository, CurrencyRepository currencyRepository) {
+        this.bankRepository = bankRepository;
+        this.currencyRepository = currencyRepository;
+    }
+
+    @Override
+    @Transactional
+    public BankDTO create(BankDTO bankDTO) {
+        Bank createdBank = bankRepository.create(BankDTO.convertToDomain(bankDTO)).orElseThrow(() -> {
+            LOGGER.debug("{} cannot be created", bankDTO);
+            throw new ResourceNotFoundException("Bank with id  is not found");
+        });
+        BankDTO convertDtoBank = BankDTO.convertToDTO(createdBank);
+        convertDtoBank.setCurrencyDTOList(new ArrayList<>());
+        if (!CollectionUtils.isEmpty(bankDTO.getCurrencyDTOList())) {
+            bankDTO.getCurrencyDTOList().forEach(currency -> {
+                Currency createdCurrency = currencyRepository.create(createdBank.getId(), convertToDomain(currency)).get();
+                CurrencyDTO convertedCurrencyDTO = convertToDTO(createdCurrency);
+                convertDtoBank.getCurrencyDTOList().add(convertedCurrencyDTO);
+            });
+        }
+
+        return convertDtoBank;
+    }
+
+    @Transactional
+    @Override
+    public BankDTO retrieveById(Long id) {
+        Bank bank = bankRepository.retrieveById(id).orElseThrow(() -> {
+            LOGGER.debug("There is no bank with id {} ", id);
+            throw new ResourceNotFoundException("Bank with id " + id + " is not found");
+        });
+
+        List<Currency> currencies = currencyRepository.retrieveAllCurrenciesByBankId(id).orElseThrow(() -> new ResourceNotFoundException("There are no currencies in this bank with id " + id));
+        BankDTO bankDTO = BankDTO.convertToDTO(bank);
+        bankDTO.setCurrencyDTOList(currencies.stream().map(CurrencyDTO::convertToDTO).collect(Collectors.toList()));
+        LOGGER.info("Bank with id {} was retrieved" + id);
```

Comment:
Use another method info(String, Object) instead of info(String)
change it to LOGGER.info("Bank with id {} was retrieved", id) without +